src/dftbp/dftb/populations.F90

```diff
@@ -533,6 +533,8 @@ subroutine getChargePerShell(qOrb, orb, species, chargePerShell, qRef)
     integer :: iAt, iSp, iSh
     integer :: nAtom, nSpin
 
+    allocate( qq(size(qorb,1),size(qorb,2),size(qorb,3)) )
```

Comment:
Lines 539 or 541 already does this through automatic allocation (as there is no dimensioning on the left), or are you getting problems with a specific compiler or need it on the heap?